### PR TITLE
Improvements for validation tests

### PR DESCRIPTION
**Pull request by nickchapman-da**
Improvements suggested by @stephencompall-DA on #10167

- prefer `Seq` to `List` when generating samples
- use  `TableDrivenPropertyChecks` / `forEvery`
- avoid `println`

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala [hunk 1]
```diff
@@ -542,45 +543,31 @@ class ValidationSpec extends AnyFreeSpec {
   //--[per tweak tests]--
 
   "Significant tweaks" - {
-    var count = 0
     significantTweaks.foreach { case (name, tweak) =>
```
Comment (S11001001) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala:546]: This also ought to be `forEvery`'d; nesting works fine. But no hurry anyway

Comment (nickchapman-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala:546]: I checked the _Table-driven property checks_ docs: https://www.scalatest.org/user_guide/table_driven_property_checks
Do you mean `forAll` ?
What's the advantage over a vanilla `foreach`?

Comment (S11001001) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala:546]: No, `forEvery`. As discussed over here https://github.com/digital-asset/daml/pull/10167#discussion_r664356959

**Pull request merged**